Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/service/decorator/KeepAliveServiceEndpointDecorator.java

```diff
@@ -0,0 +1,135 @@
+        final StringBuilder logMsgBuilder = new StringBuilder("KeepAlive statement");
+        logMsgBuilder.append(" for ");
+        logMsgBuilder.append(this.serviceName);
+        logMsgBuilder.append(" service ");
+        if (keepAliveAssertion == null) {// msg comes empty
+            final int kAFromProp = getDefaultKeepAliveFromConfig(serviceName);
+            logMsgBuilder.append("Detect Empty Keep-Alive from assertion ");
+            if (kAFromProp > 0) {
+                httpClientPolicy.setConnection(ConnectionType.KEEP_ALIVE);
+                logMsgBuilder.append("but default properties wants to have Keep Alive with duration ");
+                logMsgBuilder.append(kAFromProp);
+                logMsgBuilder.append(" seconds");
+                bindingProvider.getRequestContext().put(NhincConstants.KEEPALIVE_DURATION, kAFromProp);
+            } else {
+                httpClientPolicy.setConnection(null);
+                logMsgBuilder.append(" and default properties also wants it off.");
+            }
+        } else if (keepAliveAssertion.intValue() < 0) { // user wants it off
+            httpClientPolicy.setConnection(null);
+            logMsgBuilder.append("user explicitly wants to turn off regardless default value ");
+        } else {
+            httpClientPolicy.setConnection(ConnectionType.KEEP_ALIVE);
+            logMsgBuilder.append("user explicitly wants to turn on with duration ");
+            logMsgBuilder.append(keepAliveAssertion.intValue());
+            logMsgBuilder.append(" seconds");
+            bindingProvider.getRequestContext().put(NhincConstants.KEEPALIVE_DURATION, keepAliveAssertion);
+        }
+
+        LOG.debug(logMsgBuilder.toString());
+    }
+
+    /**
+     * Retrieve keepAliveDuration for each service in gateway properties.
+     *
+     * @param service Service name
+     * @return keepAliveDuration (in seconds) if found. Otherwise, return 0
+     */
+    public final int getDefaultKeepAliveFromConfig(final String service) {
+        int defaultKeepAlive = 0;
+        final StringBuilder keepAliveKey = new StringBuilder();
+        try {
+            keepAliveKey.append(service);
+            keepAliveKey.append(".");
+            keepAliveKey.append(KEEP_ALIVE);
+            final String sValue = PropertyAccessor.getInstance().getProperty(NhincConstants.GATEWAY_PROPERTY_FILE,
+                    keepAliveKey.toString());
+            if (NullChecker.isNotNullish(sValue)) {
+                defaultKeepAlive = Integer.parseInt(sValue);
+            }
+        } catch (final PropertyAccessException ex) {
+            LOG.warn("Error occurred reading property value from config file ({}): {}", service,
+                    ex.getLocalizedMessage(), ex);
+        } catch (final NumberFormatException nfe) {
+            LOG.warn("Error occurred converting property value from config file ({}): {}", keepAliveKey,
+                    nfe.getLocalizedMessage(), nfe);
+        }
```

Comment:
On Exception, consider logging the default value that will be used (or alternatively, a finally statement with a trace output statement containing the defaultKeepAlive value to be returned).